backend/src/org/akvo/dash/transformation.clj

```diff
@@ -1,225 +1,157 @@
-;;                    )))
-
-;; (-> [{:title "Name"} {:title "Age"}]
-;;     (update (combine (*> each)
-;;                    (conditionally #(= "Age" (:title %))))
-;;             #(assoc % :type "STRING")))
-
-;; (-> [{:title "Name" :type "STRING"} {:title "Age" :type "STRING"}]
-;;     (update (combine (*> each)
-;;                      (conditionally #(= "Age" (:title %))))
-;;             #(assoc % :type "NUMBER")))
-
-;; (pprint
-;;  (-> {:columns [{:title "Name" :type "STRING"} {:title "Age" :type "STRING"}]}
-;;      (update (*> (in [:columns])
-;;                  each
-;;                  (conditionally #(= "Age" (get % :title ))))
-;;              #(assoc % :type "NUMBER"))))
-
-;; (pprint
-;;  (-> {:columns [{"title" "Name" "type" "STRING"} {"title" "Age" "type" "STRING"}]}
-;;      (update (*> (in [:columns])
-;;                  each
-;;                  (conditionally #(= "Age" (get % "title"))))
-;;              #(assoc % "type" "NUMBER"))))
-
-;; (defn update-column-type
-;;   "Update column type to new-type on column with columns-title."
-;;   [data column-title new-type]
-;;   (-> data
-;;       (update (*> (in [:columns])
-;;                   each
-;;                   (conditionally #(= column-title (get % :title))))
-;;               #(assoc % column-title column-type))))
-
-;; (update-column-type
-;;  {:columns [{:title "Name" :type "STRING"} {:title "Age" :type "STRING"}]}
-;;  "Age"
-;;  "NUMBER"
-;;  )
-
-
-;; (defn coerce-column-view
-;;   ""
-;;   [data column-title type]
-;;   (-> data
-;;      (view (*> (in [:columns])
-;;                each
-;;                (conditionally #(= column-title (get % :title)))
-;;                (in [:values])))))
-
-
-;; (defn coerce-column
-;;   ""
-;;   [data column-title type]
-;;   (-> data
-;;       (update (*> (in [:columns])
-;;                 each
-;;                 (conditionally #(= column-title (get % :title)))
-;;                 (in [:values]))
-;;               #(map read-string %))))
-
-;; (pprint
-;;  (coerce-column
-;;   {:columns [{:title "Name" :type "STRING" :values ["1" "2" "3"]}
-;;              {:title "Age" :type "STRING" :values ["1" "2" "3"]}]}
-;;   "Age"
-;;   "NUMBER"))
-
-
-
-
-;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-;; Playing with traversy lenses
+  (:require [clojure.java.jdbc :as jdbc]
+            [clojure.string :as str]
+            [hugsql.core :as hugsql]
+            [org.akvo.dash.transformation.engine :as engine]
+            [org.akvo.dash.util :refer (squuid gen-table-name)]))
+
+;; TODO: Potential change op-spec validation `core.spec`
+
+(def ops-set (set (keys engine/available-ops)))
+(def on-error-set #{"fail" "default-value" "delete-row"})
+(def type-set #{"number" "text" "date"})
+(def sort-direction-set #{"ASC" "DESC"})
+
+(hugsql/def-db-fns "org/akvo/dash/transformation.sql")
+
+
+(defn- throw-invalid-op
+  [op-spec]
+  (throw (Exception. (str "Invalid operation " op-spec))))
```

Comment:
`clojure.core/ex-info` is nice for things like this: `(throw (ex-info "Invalid operation" op-spec))`.

Then, on catch, you can get the op-spec as data using `clojure.core/ex-data`.